lib/sync/config.go

```diff
@@ -53,23 +55,51 @@ func NewConfig(localNode *node.LocalNode,
 }
 
 func (c *Config) NewSyncer() *Syncer {
-	s := NewSyncer(c.storage, c.network, c.connectionManager, c.localNode, c.commonCfg, func(s *Syncer) {
+	f := c.NewFetcher()
+	v := c.NewValidator()
+	s := NewSyncer(f, v, c.storage, func(s *Syncer) {
 		s.poolSize = c.SyncPoolSize
-		s.fetchTimeout = c.FetchTimeout
-		s.retryInterval = c.RetryInterval
 		s.checkInterval = c.CheckBlockHeightInterval
-		s.logger = c.logger
+		s.logger = c.logger.New("module", "sync/syncer")
 	})
 
 	c.LoggingConfig()
 	return s
 }
 
+func (c *Config) NewFetcher() Fetcher {
+	f := NewBlockFetcher(
+		c.network,
+		c.connectionManager,
+		c.storage,
+		c.localNode,
+		func(f *BlockFetcher) {
+			f.fetchTimeout = c.FetchTimeout
+			f.logger = c.logger.New("module", "sync/fetcher")
+		},
+	)
+	return f
+}
+
+func (c *Config) NewValidator() Validator {
+	v := NewBlockValidator(
+		c.network,
+		c.storage,
+		c.commonCfg,
+		func(v *BlockValidator) {
+			v.prevBlockWaitTimeout = c.CheckPrevBlockInterval
+			v.logger = c.logger.New("module", "sync/validator")
+		})
+	return v
+}
+
 func (c *Config) LoggingConfig() {
 	c.logger.Info("syncer config",
+		"module", "sync/config",
```

Comment:
We can use this manner, separate by `/` for logging, setting sub module.

Reply:
https://github.com/inconshreveable/log15#the-varargs-style-is-brittle-and-error-prone-can-i-have-type-safety-please

![image](https://user-images.githubusercontent.com/12271/48896564-3ca34d00-ee8b-11e8-9639-5250517c7c76.png)


I copied that. :-> but I am ok with `module` and `submodule` :->

Reply:
Why did you remove module part, I like this manner.

Reply:
@spikeekips  `logger` in `init.go` has `module` part :-)